osquery/tables/networking/windows/process_open_sockets.cpp

```diff
@@ -6,13 +6,37 @@
  *  the LICENSE file found in the root directory of this source tree.
  */
 
+#include <string>
+#include <vector>
+
 #include <osquery/logger.h>
 #include <osquery/tables.h>
 
 #include "win_sockets.h"
 
 namespace osquery {
 namespace tables {
+namespace {
+const std::vector<std::string> kWinTcpStates = {
+    "UNKNOWN",
+    "CLOSED",
+    "LISTEN",
+    "SYN_SENT",
+    "SYN_RCVD",
+    "ESTABLISHED",
+    "FIN_WAIT1",
+    "FIN_WAIT2",
+    "CLOSE_WAIT",
+    "CLOSING",
+    "LAST_ACK",
+    "TIME_WAIT",
+    "DELETE_TCB",
+};
+
+std::string tcpStateString(const DWORD state) {
+  return state < kWinTcpStates.size() ? kWinTcpStates[state] : "UNKNOWN";
```

Comment:
nitpick: Wouldn't be better to mark a value that's most likely invalid as "INVALID"  or similar, instead of hiding it behind a "valid" value as "UNKNOWN"?
Basically the question is, wouldn't be a bug if an access out of this array is attempted?

Reply:
I modeled this after what's already there in the linux version. Basically this data structure:

https://github.com/osquery/osquery/blob/master/osquery/filesystem/linux/proc.h#L60

and this code:

https://github.com/osquery/osquery/blob/master/osquery/filesystem/linux/proc.cpp#L171